### PR TITLE
Discovery Metadata

### PR DESCRIPTION
**Pull request by Middledot**
## Summary
Implementation for interacting with guild discovery metadata

Based off of discord api docs: [discord-api-docs#1762](https://github.com/discord/discord-api-docs/pull/1762)

## Checklist

- [ ] If code changes were made then they have been tested.
- [ ] I have updated the documentation to reflect the changes.

### CODE REVIEW

**Comment by Middledot** [comment]
Docs PR has been closed

**Comment by Lulalaby** [comment]
Has been closed, like often, still we can implement it, because bots can basically use it.

File: discord/discovery.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+class DiscoveryMetadata:
+    def __init__(self, *, state: ConnectionState, guild: Guild, data):
+        self._state: ConnectionState = state
+
+        self.guild: Guild = guild
+        self.primary_category_id: int = data.get('primary_category_id')
+        self.keywords: List[str] = data.get('keywords', [])
+        self.emoji_discoverability_enabled: bool = data.get('emoji_discoverability_enabled')
+        self.partner_application: datetime.datetime = datetime.datetime.fromisoformat(data.get('partner_application_timestamp')) if data.get('partner_application_timestamp', None) is not None else None
+        self.partner_actioned: datetime.datetime = datetime.datetime.fromisoformat(data.get('partner_actioned_timestamp')) if data.get('partner_actioned_timestamp', None) is not None else None
```
Comment (plun1331) [inline discord/discovery.py:41]: `discord.utils.parse_time` can be used here instead

Example usage:
https://github.com/Pycord-Development/pycord/blob/master/discord/threads.py#L191

**Comment by Lulalaby** [comment]
Stale

**Pull request closed without merging**